internal/campaigns/service.go

```diff
@@ -95,6 +99,8 @@ mutation ApplyCampaign($campaignSpec: ID!) {
 ` + graphql.CampaignFieldsFragment
 
 func (svc *Service) ApplyCampaign(ctx context.Context, spec CampaignSpecID) (*graphql.Campaign, error) {
+	ctx = svc.decorateContext(ctx)
```

Comment:
Not a blocker: would it make sense to save the context on the `*Service` instead of passing it into every method? I don't think we call once instance of `*Service` multiple times with different contexts, so caching it might make more sense vs. having the risk of forgetting to decorate the context every time.